src/Feature/Editor/Draw.re

```diff
@@ -7,18 +7,47 @@ open Oni_Core;
 module FontIcon = Oni_Components.FontIcon;
 
 type context = {
-  transform: Reglm.Mat4.t,
+  canvasContext: CanvasContext.t,
   width: int,
   height: int,
   scrollX: float,
   scrollY: float,
   lineHeight: float,
-  fontFamily: string,
-  fontSize: int,
+  font: Revery.Font.t,
+  fontMetrics: Revery.Font.FontMetrics.t,
+  fontSize: float,
   charWidth: float,
   charHeight: float,
 };
 
+let createContext =
+    (
+      ~canvasContext,
+      ~width,
+      ~height,
+      ~scrollX,
+      ~scrollY,
+      ~lineHeight,
+      ~editorFont: EditorFont.t,
+    ) => {
+  let font = Revery.Font.load(editorFont.fontFile) |> Stdlib.Result.get_ok;
```

Comment:
Shouldn't `EditorFont.t` contain the font instead of the font file path? This really isn't the right place to deal with font fallback and such.

Reply:
Definitely, that would be a nice simplification here.